pkgs/build-support/rust/fetchcargo.nix

```diff
@@ -42,7 +43,12 @@ stdenv.mkDerivation {
     ${cargoUpdateHook}
 
     mkdir -p $out
-    cargo vendor $out | cargo-vendor-normalise > config
+    ${if flagNoMergeSources
```

Comment:
Ideally this would not be a flag at all. Because it sounds to me like something we want to have by default.

Reply:
Afaik there is no good reason to merge the packages into a single source. Probably ~80 packages need to get their hash updated in nixpkgs and out-of-tree uses might break, if we use the flag unconditionally, but we could just run `cargo-vendor` without the flag and if the `duplicate version` error, occurs re-run it with the flag, that way everything should stay the same.

Reply:
Would re-running cause `cargo-vendor` to download all dependencies twice? Otherwise this might be not a bad idea.